usermods/Battery/usermod_v2_Battery.h

```diff
@@ -0,0 +1,783 @@
+    // calculated voltage            
+    float voltage = maxBatteryVoltage;
+    // mapped battery level based on voltage
+    int8_t batteryLevel = 100;
+    // offset or calibration value to fine tune the calculated voltage
+    float calibration = USERMOD_BATTERY_CALIBRATION;
+    
+    // time left estimation feature
+    // bool calculateTimeLeftEnabled = USERMOD_BATTERY_CALCULATE_TIME_LEFT_ENABLED;
+    // float estimatedTimeLeft = 0.0;
+
+    // auto shutdown/shutoff/master off feature
+    bool autoOffEnabled = USERMOD_BATTERY_AUTO_OFF_ENABLED;
+    int8_t autoOffThreshold = USERMOD_BATTERY_AUTO_OFF_THRESHOLD;
+
+    // low power indicator feature
+    bool lowPowerIndicatorEnabled = USERMOD_BATTERY_LOW_POWER_INDICATOR_ENABLED;
+    int8_t lowPowerIndicatorPreset = USERMOD_BATTERY_LOW_POWER_INDICATOR_PRESET;
+    int8_t lowPowerIndicatorThreshold = USERMOD_BATTERY_LOW_POWER_INDICATOR_THRESHOLD;
+    int8_t lowPowerIndicatorReactivationThreshold = lowPowerIndicatorThreshold+10;
+    int8_t lowPowerIndicatorDuration = USERMOD_BATTERY_LOW_POWER_INDICATOR_DURATION;
+    bool lowPowerIndicationDone = false;
+    unsigned long lowPowerActivationTime = 0; // used temporary during active time
+    int8_t lastPreset = 0;
+
+    bool initDone = false;
+    bool initializing = true;
+
+    // strings to reduce flash memory usage (used more than twice)
+    static const char _name[];
+    static const char _readInterval[];
+    static const char _enabled[];
+    static const char _threshold[];
+    static const char _preset[];
+    static const char _duration[];
+    static const char _init[];
+    
+
+    // custom map function
+    // https://forum.arduino.cc/t/floating-point-using-map-function/348113/2
+    double mapf(double x, double in_min, double in_max, double out_min, double out_max) 
+    {
+      return (x - in_min) * (out_max - out_min) / (in_max - in_min) + out_min;
+    }
+
+    float truncate(float val, int8_t dec) 
+    {
+      float x = val * pow(10, dec);
+      float y = round(x);
+      float z = x - y;
+      if ((int)z == 5)
+        y++;
+      x = y / pow(10, dec);
+      return x;
+    }
+
+    /*
+     * Turn off all leds
+     */
+    void turnOff()
+    {
+      bri = 0;
+      stateUpdated(CALL_MODE_DIRECT_CHANGE);
+    }
+
+    /*
+     * Indicate low power by activating a configured preset for a given time and then switching back to the preset that was selected previously
+     */
+    void lowPowerIndicator()
+    {
+      if (!lowPowerIndicatorEnabled) return;
+      if (lowPowerIndicationDone && lowPowerIndicatorReactivationThreshold <= batteryLevel) lowPowerIndicationDone = false;
+      if (lowPowerIndicatorThreshold <= batteryLevel) return;
+      if (lowPowerIndicationDone) return;
+      if (lowPowerActivationTime <= 1) {
+        lowPowerActivationTime = millis();
+        lastPreset = currentPreset;
+        applyPreset(lowPowerIndicatorPreset);
+        colorUpdated(CALL_MODE_DIRECT_CHANGE);
```

Comment:
`applyPreset()` is asynchronous in 0.14 and above and does not need `colorUpdated()`.

Reply:
👍